File: day17/day17.c
#include <stdio.h>

int main() {
	long long idx = 0;
	int y = 0;
	for (int x = 1; x < 50000001; ++x) {
		/* idx = ((idx + 337) % x) + 1*/
		idx += 337; 
		idx %= x;
		if (!(idx)) {
			y = x;
		}
		++idx;
	}
	printf("%d\n", y);
}

File: day17/day17.hs
f :: Int -> Int -> Int -> Int -> Int
f idx buflen num buf1
  | num > 50000000 = buf1
  | otherwise = 
      let newidx = (idx + 337) `rem` buflen in
      f (newidx+1) (buflen+1) (num+1) (if newidx == 0 then num else buf1)

main = print $ f 0 1 1 0
